WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -0,0 +1,1047 @@
+        
+        let defaultTextAttributes = WPStyleGuide.defaultSearchBarTextAttributes(UIColor.whiteColor())
+        
+        UITextField.appearanceWhenContainedInInstancesOfClasses([UISearchBar.self, self.dynamicType]).attributedPlaceholder = attrPlacholderText
+        
+        UITextField.appearanceWhenContainedInInstancesOfClasses([UISearchBar.self, self.dynamicType]).defaultTextAttributes = defaultTextAttributes
+    }
+    
+    func configureSearchWrapper() {
+        searchWrapperView.backgroundColor = WPStyleGuide.wordPressBlue()
+    }
+    
+    func propertiesForAnalytics() -> [String:AnyObject] {
+        var properties = [String:AnyObject]()
+        
+        properties["type"] = postTypeToSync()
+        properties["filter"] = currentPostListFilter().title
+        
+        if let dotComID = blog.dotComID {
+            properties[WPAppAnalyticsKeyBlogID] = dotComID
+        }
+        
+        return properties
+    }
+    
+    // MARK: TableViewHandler Delegate Methods
+    
+    func entityName() -> String {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func managedObjectContext() -> NSManagedObjectContext {
+        return ContextManager.sharedInstance().mainContext
+    }
+    
+    func fetchRequest() -> NSFetchRequest {
+        let fetchRequest = NSFetchRequest(entityName: entityName())
+        
+        fetchRequest.predicate = predicateForFetchRequest()
+        fetchRequest.sortDescriptors = sortDescriptorsForFetchRequest()
+        fetchRequest.fetchBatchSize = self.dynamicType.postsFetchRequestBatchSize
+        fetchRequest.fetchLimit = Int(numberOfPostsPerSync())
+        
+        return fetchRequest
+    }
+  
+    func sortDescriptorsForFetchRequest() -> [NSSortDescriptor] {
+        // Ascending only for scheduled posts/pages.
+        let ascending = currentPostListFilter().filterType == .Scheduled
+        
+        let sortDescriptorLocal = NSSortDescriptor(key: "metaIsLocal", ascending: false)
+        let sortDescriptorImmediately = NSSortDescriptor(key: "metaPublishImmediately", ascending: false)
+        let sortDescriptorDate = NSSortDescriptor(key: "date_created_gmt", ascending: ascending)
+        
+        return [sortDescriptorLocal, sortDescriptorImmediately, sortDescriptorDate]
+    }
+    
+    func updateAndPerformFetchRequest() {
+        assert(NSThread.isMainThread(), "AbstractPostListViewController Error: NSFetchedResultsController accessed in BG")
+        
+        var predicate = predicateForFetchRequest()
+        let sortDescriptors = sortDescriptorsForFetchRequest()
+        let fetchRequest = tableViewHandler.resultsController.fetchRequest
+        
+        // Set the predicate based on filtering by the oldestPostDate and not searching.
+        let filter = currentPostListFilter()
+        
+        if let oldestPostDate = filter.oldestPostDate where isSearching() == false {
```

Comment:
Explanation: during the migration, and for some reason probably related with my code breaking Xcode's parser... it was asking me to explicitly use `== false`.